third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2959,19 +2962,23 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                         .add_used_spec_fun(module_id.qualified(spec_fun_id));
                     self.called_spec_funs.insert((module_id, spec_fun_id));
                 }
-                self.add_conversions(cand, &instantiation, &mut translated_args);
+                let translated_args = self.add_conversions(cand, &instantiation, translated_args);
                 let specialized_expected_type = self.subs.specialize(expected_type);
-                let mut call_exp = ExpData::Call(id, oper, translated_args).into_exp();
+                let call_exp = ExpData::Call(id, oper, translated_args).into_exp();
                 // Insert freeze for the return value
-                if result_type.is_tuple() && specialized_expected_type.is_tuple() {
-                    if let Tuple(result_tys) = result_type {
+                let call_exp = if result_type.is_tuple() && specialized_expected_type.is_tuple() {
+                    if let Tuple(ref result_tys) = result_type {
```

Comment:
It seems more common to use `if let Tuple(result_tys) = &result_type {` which I think has the same result.

Reply:
And again, you could eliminate some redundant `else` branches with a tuple:
```
if let (Tuple(result_tys), Tuple(expected_tys)) = (&result_type, specialized_expected_type) {
    ...
}
```